ui/hooks/gasFeeInput/useTransactionFunction.test.js

```diff
@@ -90,6 +90,31 @@ describe('useMaxPriorityFeePerGasInput', () => {
     });
   });
 
+  it('invokes action updateTransaction with 10% increased max priority fee and medium fee + 10% when updateTransactionToTenPercentIncreasedGasFee callback is invoked while original priority fee is 0', async () => {
+    const mockUpdateGasFees = jest
+      .spyOn(Actions, 'updateTransactionGasFees')
+      .mockImplementation(() => ({ type: '' }));
+
+    const { result } = renderUseTransactionFunctions({
+      transaction: {
+        userFeeLevel: CUSTOM_GAS_ESTIMATE,
+        txParams: { maxFeePerGas: '0x5028', maxPriorityFeePerGas: '0x0' },
```

Comment:
[completely optional]:
If we use a conversion method such as:

```javascript
function hexStringFromDec(value) {
  return new Numeric(value, 10).toBase(16).toString()
}
```

Then we can write this as

```javascript
const initialFee = 20520;

transaction: {
        userFeeLevel: CUSTOM_GAS_ESTIMATE,
        txParams: {
          maxFeePerGas: hexStringFromDec(initialFee),
          maxPriorityFeePerGas: hexStringFromDec(0)
        }
},
```

and change the assertion to be explicit about the 10% increase:

```javascript 
    expect(mockUpdateGasFees).toHaveBeenCalledWith(undefined, expect.objectContaining({
      estimateSuggested: expect.any(String),
      estimateUsed: 'custom',
      gas: expect.any(String),
      gasLimit: expect.any(String),
      maxFeePerGas: hexStringFromDec(initialFee * 1.1),
      maxPriorityFeePerGas: expect.any(String),
      userEditedGasLimit: undefined,
      userFeeLevel: expect.any(String),
  })
```

The idea is to assert as much as possible while using variables to make numbers such as `0x582c` more readable in the context of the test.

If the numbers can't be "explained" with variables, that may indicate they don't need to be matched more specifically for that assertion than with `expect.any(String)` or `expect.stringMatching(HEX_REGEX)` with `const HEX_REGEX = /0[xX][0-9a-fA-F]+/;`

Reply:
If we are going to use Numeric it is also valid to do:

```js
const x = new Numeric(value, 10);
const tenPercent = new Numeric('.10', 10); 

// later...
const transaction = {
  txParams: {
    maxFeePerGas: x.toHexString(),
  }
}

// later
expect(..).toHaveBeenCalledWith(., expect.objectContaining({
  ....
  maxFeePerGas: x.times(tenPercent).toHexString();
})
```

I am advocating for getting people comfortable working with numerics directly instead of creating intermediary methods.

Reply:
@brad-decker `toHexString` is not a method on `Numeric`?

Reply:
@brad-decker @pedronfigueiredo I have begun to collect these suggestions in a follow up PR https://github.com/MetaMask/metamask-extension/pull/17580

I'd like to merge this one quickly as it is a release blocker, but then we can clean up these tests in that PR

Reply:
100%, LGTM